arches_component_lab/views/api/card_x_node_x_widget.py

```diff
@@ -84,16 +85,42 @@ def get(self, request, graph_slug, nodegroup_alias):
         if arches_version >= (8, 0):
             card_x_node_x_widgets_query &= Q(node__source_identifier_id__isnull=True)
 
-        card_x_node_x_widgets = (
+        saved_widget_queryset = (
             models.CardXNodeXWidget.objects.filter(card_x_node_x_widgets_query)
-            .select_related()  # Eagerly load _all_ related objects
+            .select_related()
             .order_by("sortorder")
         )
 
         datatype_factory = DataTypeFactory()
-        data = [
-            serialize_card_x_node_x_widget(widget, datatype_factory)
-            for widget in card_x_node_x_widgets
-        ]
 
-        return JSONResponse(data)
+        saved_widgets_by_node_id = {
+            saved_widget.node_id: saved_widget for saved_widget in saved_widget_queryset
+        }
+
+        node_queryset = models.Node.objects.filter(
+            graph__slug=graph_slug, nodegroup__node__alias=nodegroup_alias
+        )
+        if arches_version >= (8, 0):
+            node_queryset = node_queryset.filter(source_identifier=None)
+
+        widget_instances = []
+        for node in node_queryset:
+            if node.pk in saved_widgets_by_node_id:
```

Comment:
I don't think you need to go back to the db for those nodes.  You used select related and should already have them.
```suggestion
        node_set = [widget.node for widget in saved_widget_queryset]

        if arches_version >= (8, 0):
            node_set = list(filter(lambda node: node.source_identifier is None, node_set))

        widget_instances = []
        for node in node_set:
            if node.pk in saved_widgets_by_node_id:
```